### PR TITLE
Added docs on staging app with OpenShift registry

### LINKED ISSUE #547: Add documentaion for using OpenShift registry on ADB
Refer: https://github.com/projectatomic/adb-atomic-developer-bundle/issues/372

Please check the below command output and you can see `export DOCKER_REGISTRY=hub.openshift.adb` which can used by user to push docker images. But we need to add more documentation about how it can be used.

```
[lmohanty@LalatenduM-laptop centos-openshift-setup]$ vagrant service-manager env

# docker env:
# Set the following environment variables to enable access to the
# docker daemon running inside of the vagrant virtual machine:
export DOCKER_HOST=tcp://192.168.121.67:2376
export DOCKER_CERT_PATH=/home/lmohanty/vagrantboxes/adb/adb-atomic-developer-bundle/components/centos/centos-openshift-setup/.vagrant/machines/default/libvirt/docker
export DOCKER_TLS_VERIFY=1
export DOCKER_API_VERSION=1.22

# openshift env:
# You can access the OpenShift console on: https://192.168.121.67:8443/console
# To use OpenShift CLI, run: oc login https://192.168.121.67:8443
export OPENSHIFT_URL=https://192.168.121.67:8443
export OPENSHIFT_WEB_CONSOLE=https://192.168.121.67:8443/console
export DOCKER_REGISTRY=hub.openshift.adb
```

### PR DESCRIPTION
**Pull request by thatdocslady**
Fixes #547 

This topic describes the use-case of deploying an application locally by hooking into the OpenShift Docker registry. This step can be performed after testing the app and before pushing to production. 

Created a separate topic based on a modular approach to the docs and to allow for future porting/reusing the content as needed based on scenarios.

### CODE REVIEW

**Comment by thatdocslady** [comment]
@LalatenduMohanty @praveenkumar @Preeticp please review and comment, feel free to tag other stakeholders as needed :)

File: docs/staging.adoc [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+. Build the Docker image as usual.
+
+. Make sure you are logged in to OpenShift.
++
+----
+$ oc login 10.1.2.2:8443 -u openshift-dev -p devel
+----
+
+. Tag the image against the OpenShift registry with the hostname that you retrieved
+from the `service-manager env` command.
```
Comment (praveenkumar) [inline docs/staging.adoc]: vagrant service-manager env command.

File: docs/staging.adoc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+You can prepare your application for production locally with the internal OpenShift
+Docker registry before you push your containerized application to production on OpenShift.
+
+The scenario assumes the following parameters:
+
+- OpenShift registry endpoint is *hub.openshift.adb*
+- OpenShift instance URL is *https://10.1.2.2:8443*
+- Username/password credentials is *openshift-dev/devel*
+- Application name is *my-app*
+
```
Comment (Preeticp) [inline docs/staging.adoc:13]: Would it be a good idea to add ‘sample-project’ is the name of the ADB default project, in the assumed parameters section?

Comment (praveenkumar) [reply docs/staging.adoc:13]: That's also a good idea. :+1:

File: docs/staging.adoc [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+# You can access the OpenShift console on: https://10.1.2.2:8443/console
+# To use OpenShift CLI, run: oc login https://10.1.2.2:8443
+
+export OPENSHIFT_URL=https://10.1.2.2:8443
+export OPENSHIFT_WEB_CONSOLE=https://10.1.2.2:8443/console
+export DOCKER_REGISTRY=hub.openshift.adb
+----
++
+You can also run the following command to configure your shell.
++
```
Comment (Preeticp) [inline docs/staging.adoc]: This is part of the code block which is returned when user runs the `$ vagrant service-manager env openshift` command.

**Comment by thatdocslady** [comment]
@LalatenduMohanty @praveenkumar @Preeticp I implemented your feedback, thanks! Please ack/merge if ok.

**Comment by Preeticp** [comment]
LGTM. @thatdocslady really liked the idea of the paragraph on the assumed parameters, makes it easy for the user to understand.

**Comment by praveenkumar** [comment]
LGTM :+1:

File: docs/staging.adoc [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+- OpenShift registry endpoint is *hub.openshift.adb*
+- OpenShift instance URL is *https://10.1.2.2:8443*
+- Username/password credentials is *openshift-dev/devel*
+- Default ADB project name is *sample-project*
+- Application name is *my-app*
+
+This scenario is intended to illustrate how to test an application that isn't already
+pushed to a public instance of OpenShift and that is not accessible by OpenShift
+`build config`.
+
```
Comment (LalatenduMohanty) [inline docs/staging.adoc:17]: @thatdocslady This point about `build config` is not clear to me.  What are we trying to communicate here?

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: docs/staging.adoc (hunk 4 above)
Comment (praveenkumar) [reply docs/staging.adoc:17]: This scenario is intended to illustrate how to test an application that isn’t already pushed to a public instance of OpenShift and that is not accessible by OpenShift build config. ==> This scenario is intended to illustrate how to test an application docker image to OpenShift platform (running in ADB box) which is available on your host before push it to a public instance of OpenShift.

@thatdocslady @LalatenduMohanty Does that make sense?

Comment (thatdocslady) [reply docs/staging.adoc:17]: @praveenkumar yes that's more clear, @LalatenduMohanty this text was something I took from the developer blog that you linked originally, will fix and update the PR.

**Comment by LalatenduMohanty** [comment]
LGTM

**Pull request merged**